File: Percept.cc
// Percept.cc

#include <iostream>
#include "Percept.h"

Percept::Percept ()
{
	Stench = false;
	Breeze = false;
	Glitter = false;
	Bump = false;
	Scream = false;
}

Percept::Percept (bool stench, bool breeze, bool glitter, bool bump, bool scream)
{
	Stench = stench;
	Breeze = breeze;
	Glitter = glitter;
	Bump = bump;
	Scream = scream;
}

Percept::Percept (const Percept& percept)
{
	Stench = percept.Stench;
	Breeze = percept.Breeze;
	Glitter = percept.Glitter;
	Bump = percept.Bump;
	Scream = percept.Scream;
}

Percept& Percept::operator= (const Percept& percept)
{
	if (this == &percept)
	{
		return *this;
	}
	Stench = percept.Stench;
	Breeze = percept.Breeze;
	Glitter = percept.Glitter;
	Bump = percept.Bump;
	Scream = percept.Scream;
	return *this;
}

void Percept::SetPercept (bool stench, bool breeze, bool glitter, bool bump, bool scream)
{
	Stench = stench;
	Breeze = breeze;
	Glitter = glitter;
	Bump = bump;
	Scream = scream;
}

void Percept::Print ()
{
	cout << "[Stench=" << Stench << ","
		 << "Breeze=" << Breeze << ","
		 << "Glitter=" << Glitter << ","
		 << "Bump=" << Bump << ","
		 << "Scream=" << Scream << "]" << endl;
}

File: Percept.py
# Percept.py

class Percept:
	def __init__(self):
		self.set_percept(False, False, False, False, False)
	
	def __str__(self):
		s = "[Stench=" + str(self.stench)
		s += ",Breeze=" + str(self.breeze)
		s += ",Glitter=" + str(self.glitter)
		s += ",Bump=" + str(self.bump)
		s += ",Scream=" + str(self.scream) + "]"
		return s
		
	def set_percept(self, stench, breeze, glitter, bump, scream):
		self.stench = stench
		self.breeze = breeze
		self.glitter = glitter
		self.bump = bump
		self.scream = scream

